src/citrine/resources/dataset.py

```diff
@@ -380,6 +383,97 @@ def gemd_batch_delete(
         return self.gemd.batch_delete(id_list, timeout=timeout, polling_delay=polling_delay)
 
 
+class TeamDatasetCollection(Collection[Dataset]):
```

Comment:
@kroenlein I was temped to use the `DatasetCollections` for the team-based ones and make it parametrizable, that I can't introduce any breaking change over there, so I'm creating a new collection here.

Reply:
Can you detail what would break by making the project_id potentially None?  If the route is going away, all traffic should be going through the team-based route.

Reply:
If you are thinking of adding the `team_id` as an optional parameter and making the `project_id` to be optional for the constructor DatasetCollection, I'd say that some people might be using positional parameters for constructing it. Or... what do you have in mind?

Reply:
People aren't using either of those patterns.  The general usage is 
```
citrine = Citrine()
team = citrine.teams.get('xxx')
project = team.projects.get('xxx')
dataset = project.datasets.get('xxx')
```
Because of some weird interface choices, doing this just with constructors is actually remarkably awkward.

Reply:
Got it, that helps me a lot